project-management-infrastructure/src/main/java/life/qbic/projectmanagement/infrastructure/organisations/RorApi.java

```diff
@@ -0,0 +1,152 @@
+      this.organisationApiEndpoint = URI.create(organisationApiEndpoint);
+      this.apiClientId = Objects.requireNonNull(apiClientId);
+    }
+
+    public static final class RorEntryV2 implements RorEntry {
+
+      @JsonProperty("id")
+      String id;
+
+      @JsonProperty("names")
+      List<OrganisationName> names;
+
+      public static class OrganisationName {
+
+        @JsonProperty("lang")
+        String language;
+
+        @JsonProperty("types")
+        List<String> types;
+
+        @JsonProperty("value")
+        String value;
+
+        public String getLanguage() {
+          return language;
+        }
+
+        public List<String> getTypes() {
+          return types;
+        }
+
+        public String getValue() {
+          return value;
+        }
+      }
+
+      public String getId() {
+        return id;
+      }
+
+      @Override
+      public String getDisplayedName() {
+        return names.stream().filter(name -> name.getTypes().contains("ror_display")).findFirst()
+            .map(OrganisationName::getValue)
+            .orElseThrow();
+      }
+    }
+
+    @Override
+    public Optional<RorEntry> find(String rorId) {
+
+      try (HttpClient client = HttpClient.newBuilder().version(Version.HTTP_2)
+          .followRedirects(Redirect.NORMAL).connectTimeout(
+              Duration.ofSeconds(10)).build();) {
+
+        URI requestUri = organisationApiEndpoint.resolve(rorId);
+
+        var queryBuilder = HttpRequest.newBuilder()
+            .uri(requestUri)
+            .header("Content-Type", "application/json")
+            .header("Client-Id", apiClientId)
+            .GET();
+
+        var rorQuery = queryBuilder.build();
+        var result = client.send(rorQuery, BodyHandlers.ofString());
+        //If a valid RoRId was provided but the ID does not exist we fail
+        if (result.statusCode() == 404) {
+          log.warn(
+              "Provided Organisation ROR id: %s was not found via API call to %s".formatted(rorId,
+                  requestUri));
+          return Optional.empty();
+        }
+        //If a valid RoRId was provided but the ID does not exist we fail
+        else if (result.statusCode() != 200) {
+          log.warn(
+              ("Unexpected error retrieving an organization with ROR id %s. API call to %s").formatted(
+                  rorId,
+                  requestUri));
+          return Optional.empty();
+        }
+        RorEntry rorEntry = new ObjectMapper().configure(
+                DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+            .readValue(result.body(), RorEntryV2.class);
+        return Optional.of(rorEntry);
+      } catch (IOException | InterruptedException e) {
+        log.error("Finding ROR entry failed for organisation: %s".formatted(rorId), e);
+        /* Clean up whatever needs to be handled before interrupting  */
+        Thread.currentThread().interrupt();
```

Comment:
Only propagate the thread interrupten when an interrupted exception is thrown. Otherwise you signal the current thread to stop on IOExceptions (e.g. network) which can cause nasty side effects in our application. 

So either you have two separate catch blocks here or you check the `Thread.current().isInterrupted()` flag.